packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts

```diff
@@ -63,14 +64,28 @@ export function getSPOAndGraphRequestIdsFromResponse(headers: { get: (id: string
     return additionalProps;
 }
 
+export interface IFacetCodes {
+    facetCodes?: string[];
+ }
+
+ function parseFacetCodes(response): string[] {
+    const stack: string[] = [];
+    let error = response.error;
+    while (error !== undefined) {
+        stack.push(error.code);
```

Comment:
It would be great for this code to be more protective (i.e. use defense in depth) - we are operating with server response and it can be anything (i.e. there is no guarantee that it will be error facet).

So I'd check that error is object (check for null as unfortunately _typeof null === "object"_ in JS)
Check that it actually has code property and it is a string (I assume it should be a string from ODSP spec? if not sure, we should clarify with them).

Reply:
It would be great to add UTs that will call this function with various mal-formatted structures (including non-JSON payload) and validate that it does not crash and return some reasonable outcome.